scripts/src/main/resources/scripts/command/az

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env bash
+
+# autocompletion list
+if [ "${1}" = "shortlist" ]
+then
+  if [ -z "${2}" ]
+  then
+    echo "setup help"
+  fi
+  exit
+fi
+
+if [ -n "${DEVON_IDE_TRACE}" ]; then set -vx; fi
+# shellcheck source=scripts/functions
+source "$(dirname "${0}")"/../functions
+
+function doSetup() {
+  if doIsAzureInstalled
+  then
+    if [ "${1}" != "silent" ] && ! doIsQuiet
+    then
+      doEcho "Azure CLI is already installed at $(command -v az)"
+    fi
+  else
+    doRequireNotBatch
+    if doIsWindows
+    then
+      # Get leatest release
+      if [ -z "${AZURECLI_VERSION}" ]
+      then
+        doEcho "Getting latest release..."
+        AZURECLI_VERSION=$(curl -s https://api.github.com/repos/Azure/azure-cli/releases/latest | awk -F ":" '/tag_name/ {print $2}'| awk -F "\"" '{print $2}' | awk -F "-" '{print $3}')
+      fi
+      local version="${AZURECLI_VERSION}"
+      doDownload "-" "" "az" "${version}"
+      # Install Azure CLI for Windows
+      doEcho "Installing Azure CLI for Windows..."
+      modifyPathStructure
+      installOnWindows
+    else doFail "Sorry, Azure CLI installation support is not yet implemented for your OS. Please install manually or help devonfw-ide to support it for your OS by contributing a pull-request."
+    fi
+  fi
+}
+
+# modify a filepath for powershell path syntax e.g. /c/... -> c:/...
+function modifyPathStructure(){
+  modifiedPath=${DEVON_DOWNLOAD_DIR}
+  partition=$(echo "${modifiedPath}" | cut -f2 -d'/'):'/' # -> <partition>:/
+  modifiedPath=${partition}$(echo "${modifiedPath}" | cut -f3- -d'/')
+  modifiedPath="${modifiedPath////\\}"
+}
+
+# Install Azure CLI for Windows via Powershell Command
+function installOnWindows(){ 
+  if doIsPwshInstalled
+  then
+    powershell.exe -Command "Start-Process msiexec.exe -verb runas -Wait -ArgumentList '/I ${modifiedPath}\az-${version}-windows.msi /QB-!'" &> /dev/null || exit_code=$?
+    if [[ -z "${exit_code}" ]]
+    then 
+      setConfigDir
+      doEcho "Azure CLI has been installed and is ready to use. \n\n******** IMPORTANT ******** \nRestart the console to use Azure CLI regular." 
+    else
+      read -r -p "You must run the installation with administration rights. Would you like to start the installation again? (Y/n)" answer 
+      case ${answer:0:1} in 
+        y|Y )
+          doEcho "\nInstalling Azure CLI for Windows..."
+          exit_code=""
+          installOnWindows
```

Comment:
Assuming the user decides to rerun the installation after it failed due to the lack of admin rights.
This will not change anything so the same error will happen as it runs again without admin rights, correct?
If yes, this entire question and repeating does not make sense to me.

Reply:
The variable really makes no sense.

Reply:
> Assuming the user decides to rerun the installation after it failed due to the lack of admin rights.
This will not change anything so the same error will happen as it runs again without admin rights, correct?

I still do not see this addressed. Do I miss something? What is the point in rerunning the installation without any change again if exit code was not 0 (succcess)? And the assumption that the admin permissions were causing the error might also be wrong. So if admin rights are required, the powershell process needs to be launched in a specific way granting admin rights and if that is done, there is IMHO no sense in retrying.

Or is this meant as following:
The PS is already launched with admin rights what renders a UAC popup dialog. If the user clicks on cancel you are telling that admin rights are required and rerun it so the user has to accept UAC. Is that what you intended?
Then it might be OK but shouldn't the user rerun the `setup` if he accidentally clicked cancel?
If he for whatever reason intended to cancel, will he be annoyed by devonfw-ide asking him forever, making it hard to abort?